webapp/app.py

```diff
@@ -100,9 +110,49 @@ def api():
     Show the static api page
     """
 
+    base_url = "https://raw.githubusercontent.com/maas/maas-openapi-yaml/main"
+    yaml_file = "openapi2.yaml"
+
+    try:
+        definition = request.urlopen(f"{base_url}/{yaml_file}")
+    except error.HTTPError or error.URLError:
+        # Fall back to local yaml file if hosted yaml is unreachable
+        definition = open(yaml_file)
+    loaded_definition = load(definition, Loader)
+    tagged_definition = {}
+
+    for endpoint in loaded_definition["paths"]:
+        for method in loaded_definition["paths"][endpoint]:
+            if method != "parameters":
+                try:
+                    if {
+                        endpoint: loaded_definition["paths"][endpoint]
+                    } not in tagged_definition[
+                        loaded_definition["paths"][endpoint][method]["tags"][0]
+                    ]:
+                        tagged_definition[
+                            loaded_definition["paths"][endpoint][method][
+                                "tags"
+                            ][0]
```

Comment:
Do you think turning this into a named variable before using it here could make this code clearer? All these nested dictionaries make it quite hard to read.

Reply:
You're right. I think there's a lot of that I could do in the HTML as well, trying to reference everything was no easy task. I can crack on with that tomorrow to get it to a more readable state